automation/default_browser_params.json

```diff
@@ -2,7 +2,7 @@
     "extension_enabled": true,
     "cookie_instrument": false,
     "js_instrument": false,
-    "js_instrument_modules": "fingerprinting",
+    "js_instrument_modules": ["fingerprinting"],
```

Comment:
Don't forget to retain support for crawler env vars (strings only) here

Reply:
Indeed. Thanks for the reminder.I bumped into that when I was running crawls. [This](https://github.com/mozilla/OpenWPM/commit/361982e4d58c765621d87593444ddcc1199ab58b#diff-ee054b4b797f6c2a5743d2654764a8ed) is the change I made to `crawler.py`. I should add this to this PR and will need to add documentation about structuring the `crawl.yaml` probably to openwpm-crawler. I was actually please with how that ends up looking:

```yaml
        - name: NAVIGATION_INSTRUMENT
          value: '1'
        - name: JS_INSTRUMENT
          value: '1'
        - name: CALLSTACK_INSTRUMENT
          value: '1'
        - name: JS_INSTRUMENT_MODULES
          value: >
            [
                "AbortController",
                "window.navigator",
                "fingerprinting"
            ]
        - name: DWELL_TIME
          value: '15'
        - name: TIMEOUT
          value: '30'
```

Reply:
Having an ability to use JSON to specify instrumentation configuration / options is a great addition in general. I'd love to see one more change here though - don't re-use JS_INSTRUMENT_MODULES. Add a separate config option called something like JS_INSTRUMENT_ATTRIBUTES or similar. Separation of concerns, allowing for JS_INSTRUMENT_MODULES and JS_INSTRUMENT_ATTRIBUTES to live separate lives in the codebase.

Reply:
I wouldn't say this PR re-uses JS_INSTRUMENT_MODULES but completely redefines it. That may or may not be good.

Can you go into a bit more detail so I can understand what you're proposing: 
* JS_INSTRUMENT_MODULES and JS_INSTRUMENT_ATTRIBUTES are different concepts and where they would come back together for a uniifed

Reply:
If it completely redefines it and doesn't keep anything related to configuring which js "modules" are to be used, a rename may be more appropriate. Maybe just rename to _ATTRIBUTES, or _SPEC?

Reply:
This would also solve my bad choice of _MODULES in the first place, since it is confusing that OpenWPM docs talks about the different instruments as modules, and then there is a _MODULES config option for a module :)

Reply:
Naming things is not my strong suit. @englehardt thoughts?

Reply:
I suggest renaming the browser param name as well.

Reply:
Using _SETTINGS is confusing as well, since _MODULES (if kept) is also a js instrument configuration / setting. Ideally, something specific to what the setting does should also be its name, eg if it contains the spec for js instrument to use, _SPEC could be good.

Reply:
Sarah has removed the notion of a "module" and replaced it with a "shortcut" (i.e., a predefined set of APIs to instrument).

Reply:
_spec, _settings, _config, ....

Reply:
> _spec, _settings, _config, ....

You are right. _spec is no better. What about _apis? JS_INSTRUMENT_APIS

Reply:
Let's use `_SETTINGS`. We don't have a distinction between modules and non-modules so it covers everything nicely.

Reply:
Resolved in 8f593c8